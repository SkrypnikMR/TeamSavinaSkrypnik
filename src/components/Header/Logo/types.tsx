import React from 'react';

import { ITheme } from 'src/components/App/types';

export interface IDiv extends ITheme{
    display: string;
    flexDirection: string;
    justifycontent: string;
    alignItems: string;
    width: string;
    fontFamily: string;
    color: string;
    fontSize: string;
    height: string;
    marginTop: string;
    cursor: string;
}
