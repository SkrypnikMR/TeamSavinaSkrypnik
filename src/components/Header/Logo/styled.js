import styled from 'styled-components';

export const StLogo = styled.div`
    display:flex;
    justify-content: space-evenly;
    align-items: center;
    width:30%;
    font-family: 'Play', sans-serif;
    color:white;
        h1{
            font-size: 30px;
        }
        img{
            height: 8vh;
        }
    cursor: pointer;
`;
