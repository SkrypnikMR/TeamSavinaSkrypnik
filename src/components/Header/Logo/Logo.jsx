import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StLogo } from './styled';

const Logo = ({ history, setValue }) => {
    const handleClick = () => {
        const { pathname } = history.location;
        if (pathname === '/' || pathname === '/registration') return history.push('/');
        setValue({ name: 'init', value: true });
        history.push('/chat');
    };
    const { t } = useTranslation();
    return (
        <StLogo onClick={handleClick}>
            <img src="./public/assets/images/logo.png" />
            <h1>{t('logo')}</h1>
        </StLogo>
    );
};

Logo.propTypes = {
    history: PropTypes.object,
    setValue: PropTypes.func,
};

export default Logo;
