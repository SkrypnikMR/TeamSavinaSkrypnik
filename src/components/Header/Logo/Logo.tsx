import React from 'react';
import { useTranslation } from 'react-i18next';
import Button from 'src/components/UI/Button';
import { StLogo } from './styled';
import { LOGO } from '../../UI/baseLayout';

const Logo = ({ history }) => {
    const goToGames = () => history.push('/mainPage');
    const goToStats = () => history.push('/statistics');
    const { t } = useTranslation();
    return (
        <StLogo >
            <img src={LOGO} onClick={goToGames} />
            <div>
                <Button
                    content={t('game_zone')}
                    fontSize='20px'
                    borderRadius="0px"
                    bgColor="transparent"
                    onClick={goToGames}
                />
                <Button
                    content={t('statistics')}
                    fontSize='20px'
                    borderRadius="0px"
                    bgColor="transparent"
                    onClick={goToStats}
                />
            </div>
        </StLogo>
    );
};

export default Logo;
