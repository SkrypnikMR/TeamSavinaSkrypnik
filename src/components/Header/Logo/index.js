import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import Logo from './Logo.jsx';
import { setValue } from '../../../Store/user/actions';

const mapDispatchToProps = dispatch => ({ setValue: payload => dispatch(setValue(payload)) });

export default connect(null, mapDispatchToProps)(withRouter(Logo));
