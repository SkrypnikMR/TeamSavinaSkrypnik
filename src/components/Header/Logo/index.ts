import { withRouter } from 'react-router';

import Logo from './Logo';

export default withRouter(Logo);
