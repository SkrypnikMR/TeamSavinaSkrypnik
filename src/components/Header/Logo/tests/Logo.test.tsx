import React from 'react';
import { shallow, mount } from 'enzyme';
import Logo from '../Logo';

describe('Logo', () => {
    const props = {
        history: { push: jest.fn() },
    };
    it('Should match snapshot', () => {
        const component = shallow(<Logo {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should render 2 btns', () => {
        const component = mount(<Logo {...props} />);
        expect(component.find('Button')).toHaveLength(2);
    });
    it('Should render img', () => {
        const component = mount(<Logo {...props} />);
        expect(component.find('img')).toHaveLength(1);
    });
    it('Should cal history push with /mainPage', () => {
        const component = mount(<Logo {...props} />);
        component.find('Button').at(0).props().onClick();
        expect(props.history.push).toHaveBeenCalledWith('/mainPage');
    });
    it('Should cal history push with /statistics', () => {
        const component = mount(<Logo {...props} />);
        component.find('Button').at(1).props().onClick();
        expect(props.history.push).toHaveBeenCalledWith('/statistics');
    });
});
