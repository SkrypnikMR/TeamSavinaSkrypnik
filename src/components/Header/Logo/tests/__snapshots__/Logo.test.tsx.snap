// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Should match snapshot 1`] = `"<div class=\\"styled__StLogo-sc-1lhyjdk-0 eFYqgc\\"><img src=\\"public/assets/images/game-development.png\\"/><div><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 kFMylk\\">game_zone</button><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 qmQvK\\">statistics</button></div></div>"`;
